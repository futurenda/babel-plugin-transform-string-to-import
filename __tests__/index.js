const pluginTester = require('babel-plugin-tester')
const plugin = require('../index.js')

pluginTester({
    plugin,
    tests: {
        'transforms string expression': {
            code: `
                var a = '~iconName'
            `,
            snapshot: true
        },
        'transforms string in function': {
            code: `
                function foo() {
                    var a = '~iconName'
                }
            `,
            snapshot: true
        },
        'transforms string in jsx': {
            code: `
                function foo() {
                    return <icon name="~iconName"/>
                }
            `,
            snapshot: true
        },
        'doesn\'t generate duplicate imports': {
            code: `
                var a = '~iconName'
                var b = '~iconName'
                function foo() {
                    return <icon name="~iconName"/>
                }
            `,
            snapshot: true
        },
        'works with expressions': {
            code: `
                function foo() {
                    return <icon name={'~iconName'}/>
                }
            `,
            snapshot: true
        },
    }
})