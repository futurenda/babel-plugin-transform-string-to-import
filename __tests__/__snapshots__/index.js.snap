// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform-string-to-import doesn't generate duplicate imports: doesn't generate duplicate imports 1`] = `
"
var a = '~iconName'
var b = '~iconName'
function foo() {
    return <icon name=\\"~iconName\\"/>
}

      ↓ ↓ ↓ ↓ ↓ ↓

import _iconName from 'iconName';
var a = _iconName;
var b = _iconName;
function foo() {
    return <icon name={_iconName} />;
}
"
`;

exports[`transform-string-to-import transforms string expression: transforms string expression 1`] = `
"
var a = '~iconName'

      ↓ ↓ ↓ ↓ ↓ ↓

import _iconName from 'iconName';
var a = _iconName;
"
`;

exports[`transform-string-to-import transforms string in function: transforms string in function 1`] = `
"
function foo() {
    var a = '~iconName'
}

      ↓ ↓ ↓ ↓ ↓ ↓

import _iconName from 'iconName';
function foo() {
    var a = _iconName;
}
"
`;

exports[`transform-string-to-import transforms string in jsx: transforms string in jsx 1`] = `
"
function foo() {
    return <icon name=\\"~iconName\\"/>
}

      ↓ ↓ ↓ ↓ ↓ ↓

import _iconName from \\"iconName\\";
function foo() {
    return <icon name={_iconName} />;
}
"
`;

exports[`transform-string-to-import works with expressions: works with expressions 1`] = `
"
function foo() {
    return <icon name={'~iconName'}/>
}

      ↓ ↓ ↓ ↓ ↓ ↓

import _iconName from 'iconName';
function foo() {
    return <icon name={_iconName} />;
}
"
`;
